[sxfoo] iP

sxfoo opened the pull request.

sxfoo committed:
Rename chatbot from Duke to Carrot

sxfoo committed:
Implement chatbot basic greeting and exit functionality

sxfoo committed:
Reorganise chatbot greeting and goodbye messages into two separate methods

sxfoo committed:
Add functionality to receive userInput and echo

Level-1 Echo functionality - echos commands entered by the user, and exits when the user types the command bye.

sxfoo committed:
Add functionality to store userInput (tasks) and display tasks

Level-2 Add, List functionality. Add the ability to store whatever text entered by the user and display them back to the user when requested (when user types the keyword "list").

sxfoo committed:
Add new class: Task

sxfoo committed:
Add functionality to mark/unmark tasks

Level-3 Mark/Unmark Tasks.
Add the ability to mark tasks as done (userInput: "mark (taskIndex)"), and not done (userInput: "unmark (taskIndex)").

sxfoo committed:
Tweak the code to comply with Java Coding Standard

A-CodingStandard.
Changed the code to fit standards as described in https://se-education.org/guides/conventions/java/basic.html

sxfoo committed:
Add new nullptr exception block in mark/unmark method

sxfoo committed:
Rename Duke.java to Carrot.java to match it's new name

sxfoo committed:
Add new classes Todo, Deadline, Event and modify toString() in Task.

sxfoo committed:
Add new class CommandManager and enum CommandType

To handle userinput commands on a separate class

sxfoo committed:
Change main Carrot.java code to use CommandManager class

sxfoo committed:
Add new command HELP in CommandManager

sxfoo committed:
Add Todo, Deadline, Event, Help functionality in main

Other things: Refactor out print statements to separate functions. Changed Task[] array to use ArrayList instead of normal Array

Mmaxx15 commented on src/main/java/Carrot.java:
```diff
@@ -0,0 +1,215 @@
+
+        try {
+            changeTaskStatus(isDone, taskIndexInt);
+        } catch (NumberFormatException | IndexOutOfBoundsException | NullPointerException e) {
```
Well done for error handling

PDHung1104 replied:
Perhaps multiple ```catch``` is better

Mmaxx15 commented on src/main/java/Carrot.java:
```diff
@@ -0,0 +1,215 @@
+            "7. bye - Exit the program\n" +
+            "8. help - Show available commands";
+
+    private static void greetUser() {
```
Good naming convention(camelCase) (for all function names)

Mmaxx15 commented on src/main/java/CommandManager.java:
```diff
@@ -0,0 +1,103 @@
+    private static final Pattern DEADLINE_PATTERN =
+            Pattern.compile("deadline (?<taskDescription>.+) /(?<by>.+)", Pattern.CASE_INSENSITIVE);
+    private static final Pattern EVENT_PATTERN =
+            Pattern.compile("event (?<taskDescription>.+) /(?<from>.+) /(?<to>.+)", Pattern.CASE_INSENSITIVE);
```
Good job on keeping the line length less than 120 characters

Mmaxx15 commented on src/main/java/Carrot.java:
```diff
@@ -0,0 +1,215 @@
+        String taskIndex =
+                CommandManager.getCommandArguments(command, userInput)[0];
+        int taskIndexInt =
+                Integer.parseInt(taskIndex);
```
Can consider moving this up as it is less than 120 characters (more than 1 occurrance)

PDHung1104 commented on src/main/java/Carrot.java:
```diff
@@ -0,0 +1,215 @@
+
+public class Carrot {
+    public static void main(String[] args) {
+        String logo = "   _____                     _   \n" +
+                        "  / ____|                   | |  \n" +
+                        " | |     __ _ _ __ _ __ ___ | |_ \n" +
+                        " | |    / _` | '__| '__/ _ \\| __|\n" +
+                        " | |___| (_| | |  | | | (_) | |_ \n" +
+                        "  \\_____\\__,_|_|  |_|  \\___/ \\__|\n";
```
Nice logo

PDHung1104 commented on src/main/java/CommandManager.java:
```diff
@@ -0,0 +1,103 @@
+            return null;
+        }
+
+        // Check if userinput matches the associated pattern
+        Matcher matcher = commandPattern.matcher(userInput);
+        if (!matcher.matches()) {
+            return null;
+        }
+
+        // Return in String[], the user arguments associated with the command
+        // eg. <taskIndex> for mark, <taskDescription> && <by> for deadline,
+        String[] arguments = null;
+
+        switch (command) {
+        case MARK:
+        case UNMARK:
+            arguments = new String[1];
+            arguments[0] = matcher.group("taskIndex");
+            break;
+        case TODO:
+            arguments = new String[1];
+            arguments[0] = matcher.group("taskDescription");
+            break;
+        case DEADLINE:
+            arguments = new String[2];
+            arguments[0] = matcher.group("taskDescription");
+            arguments[1] = matcher.group("by");
+            break;
+        case EVENT:
+            arguments = new String[3];
+            arguments[0] = matcher.group("taskDescription");
+            arguments[1] = matcher.group("from");
+            arguments[2] = matcher.group("to");
+            break;
+        default:
+            return null;
+        }
+
+        return arguments;
+    }
```
Try to keep the method's LoC <= 30

PDHung1104 commented on src/main/java/Carrot.java:
```diff
@@ -0,0 +1,215 @@
+
+        // Variables to hold current userInput and associated command
+        String userInput;
+        CommandType userCommand;
+
+        // Chat input and response Loop
+        while (true) {
+            userInput = in.nextLine().trim();
+            userCommand = CommandManager.getCommandType(userInput);
+
+            switch (userCommand) {
+            case EXIT:
+                return;
+            case INVALID:
+                printInvalidCommand();
+                break;
+            case HELP:
+                printHelpCommand();
+                break;
+            case LIST:
+                printListItems();
+                break;
+            case TODO:
+                addTodoToList(userInput);
+                break;
+            case DEADLINE:
+                addDeadlineToList(userInput);
+                break;
+            case EVENT:
+                addEventToList(userInput);
+                break;
+            case MARK:
+            case UNMARK:
+                handleMarkUnmarkTask(userCommand, userInput);
+                break;
+            default:
+                break;
+            }
+        }
+    }
```
Try to shorten this method

PDHung1104 commented on src/main/java/Carrot.java:
```diff
@@ -0,0 +1,215 @@
+        System.out.println("Hello from\n" + logo);
+
+        // Chatbot functionality
+        greetUser();
+        handleUserInput();
+        sayGoodbye();
```
Good ```SLAP Hard``` practice

PDHung1104 reviewed (commented):
Overall, your code has good quality. ```SLAP Hard``` was practiced in most part of your code base. However, try to reduce the length of some methods.

sxfoo committed:
Refactor switch command handling method to reduce LOC

sxfoo committed:
Add CarrotException Class

sxfoo committed:
Revert "Add CarrotException Class"

This reverts commit c8faeb6659ae60012f2ae980a49b91fca6d8f612.

sxfoo committed:
Add CarrotException Class

sxfoo committed:
Merge branch 'branch-Level-5'

sxfoo committed:
Add support for deleting tasks from user's list

New command "delete <taskIndex>" to remove list item from list. Refactor out todo,deadline and event code to a single function.

sxfoo committed:
Add Storage Class for storing data in a textfile

Save the tasks in the hard disk automatically whenever the task list changes. Load the data from the hard disk when the chatbot starts up. Textfile saved in data/listOfTasks.txt

sxfoo committed:
Modify Carrot class to load and save data in storage

sxfoo committed:
Merge branch 'branch-Level-6'

sxfoo committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/Carrot.java

sxfoo committed:
Add information for "delete" command in help

madhanse commented on src/main/java/Carrot.java:
```diff
@@ -0,0 +1,219 @@
+        case LIST:
+            printListItems();
+            break;
+        case TODO:
```
As per coding standards, if there is no break in the case statement, '// Fallthrough' comment should be added.
![image](https://github.com/nus-cs2113-AY2324S2/ip/assets/69302316/d0675607-1289-463a-b316-e09df1c432fa)

sxfoo committed:
Extract out Ui class

sxfoo committed:
Extract TaskList to a separate class, Rename CommandManager to Parser

sxfoo committed:
Move class files to separate packages

sxfoo committed:
Extract out Command to their separate classes

sxfoo committed:
Add Find command functionality

sxfoo committed:
Add JavaDoc comments

sxfoo committed:
Merge pull request #1 from sxfoo/branch-Level-9

Add Find command functionality

sxfoo committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/parser/Parser.java
#	src/main/java/ui/Ui.java

sxfoo committed:
Merge pull request #2 from sxfoo/branch-A-JavaDoc

Add JavaDoc comments

sxfoo committed:
Fix some javadoc comments & write new help text for find command

sxfoo committed:
Add userguide

sxfoo committed:
Attempt fix README.md on website #1
